pipeline/detection.py

```diff
@@ -98,20 +201,422 @@ def run(self, *args, **kwargs):
+            temporary image, weight, and mask file will be written with
+            the same name base to the same directory, but will be
+            deleted by this routine unless do_not_cleanup is False.)  It
+            is the responsibility of the calling routine to delete this
+            temporary file.
+
+          do_not_cleanup: bool, default False
+            This routine writes some temp files with the image, weight,
+            mask, and sourcelist data in it, muchof which is probably
+            redundant with what's already written somewhere.  Normally,
+            they're deleted at the end of the routine.  Set this to True
+            to keep the files for debugging purposes.  (If tempname is
+            not None, then the sourcelist file will not be deleted even
+            if this is False; the reasoning is, if you passed a
+            tempname, it's because you needed that file, and indeed
+            extract_sources_sextractor does.)
+
+        Returns
+        -------
+          Source List
+            Has data and info already loaded
+
+        """
+        tmpnamebase = tempname
+        if tmpnamebase is None: tmpnamebase = ''.join( random.choices( 'abcdefghijklmnopqrstuvwxyz', k=10 ) )
+        tmpimage = pathlib.Path( FileOnDiskMixin.temp_path ) / f'{tmpnamebase}.fits'
+        tmpweight = tmpimage.parent / f'{tmpnamebase}.weight.fits'
+        tmpflags = tmpimage.parent / f'{tmpnamebase}.flags.fits'
+        tmpsources = tmpimage.parent / f'{tmpnamebase}.sources.fits'
+        tmpparams = tmpimage.parent / f'{tmpnamebase}.param'
+
+        # For debugging purposes
+        self._tmpimage = tmpimage
+        self._tmpweight = tmpweight
+        self._tmpflags = tmpflags
+        self._tmpsources = tmpsources
+
+        if image.data is None or image.weight is None or image.flags is None:
+            raise RuntimeError( f"Must have all of image data, weight, and flags" )
+
+        # Figure out where astromatic config files are:
+        astromatic_dir = None
+        cfg = Config.get()
+        if cfg.value( 'astromatic.config_dir' ) is not None:
+            astromatic_dir = pathlib.Path( cfg.value( 'astromatic.config_dir' ) )
+        elif cfg.value( 'astromatic.config_subdir' ) is not None:
+            astromatic_dir = pathlib.Path( CODE_ROOT ) / cfg.value( 'astromatic.config_subdir' )
+        if astromatic_dir is None:
+            raise FileNotFoundError( "Can't figure out where astromatic config directory is" )
+        if not astromatic_dir.is_dir():
+            raise FileNotFoundError( f"Astromatic config dir {str(astromatic_dir)} doesn't exist "
+                                     f"or isn't a directory." )
+
+        # TODO : make these configurable by instrument (at least!)
+        # For now we're using the astromatic defaults that everybody
+        # just uses....
+        conv = astromatic_dir / "default.conv"
+        nnw = astromatic_dir / "default.nnw"
```

Comment:
We should think about whether we want to config the pipeline to have instrument-specific behavior, or just assume you'd always run it on a single instrument, then you can just choose all of these parameters in the one config file for the whole system.

Reply:
We almost certainly will need instrument-specific behavior.  I can easily imagine us ending up needing to put together multiple different instruments, even if we don't want to.  (There will be some program that, say, taktes data on DECam at the same time as LS4, and they want us to run it all through the pipeline to compare discovered candidates, etc.)

Reply:
yes, but even then, would it not be easier to just have some machines running with config file A and other machines running config file B? 

If we are going to split off one CCD per computer / processor then I don't see that same processor running DECam one minute and then LSST on the next...